scripts/extract_operations.py

```
@@ -0,0 +1,157 @@
+    }
+    canonical = json.dumps(manifest, sort_keys=True, separators=(",", ":")).encode()
+    manifest["manifest_sha256"] = hashlib.sha256(canonical).hexdigest()
+    return manifest
+
+
+def verify_manifest(manifest: dict) -> None:
+    """Sabotage: verify manifest self-consistency."""
+    errors = []
+    for field in ("class", "schema_version", "operation_count", "operations", "manifest_sha256"):
+        if field not in manifest:
+            errors.append(f"MISSING_FIELD: {field}")
```

Ruff (0.16.3)

[warning] Use a list comprehension to create a transformed list

(PERF401)